napari/_qt/dialogs/qt_plugin_dialog.py

```diff
@@ -355,27 +409,46 @@ def __init__(
         enabled: bool = True,
         installed: bool = False,
         npe_version=1,
+        versions_conda: list = [],
+        versions_pypi: list = [],
     ):
         super().__init__(parent)
+        self.url = url
+        self._versions = {}
+        self._versions['Conda'] = versions_conda
+        self._versions['PyPI'] = versions_pypi
         self.setup_ui(enabled)
         self.plugin_name.setText(package_name)
+
+        self._populate_version_dropdown('PyPI')
         self.package_name.setText(version)
-        self.summary.setText(summary)
-        self.package_author.setText(author)
+        if summary:
+            self.summary.setText(summary)
+        if author:
+            self.package_author.setText(author)
+        self.package_author.setWordWrap(True)
         self.cancel_btn.setVisible(False)
 
-        self.help_button.setText(trans._("Website"))
-        self.help_button.setObjectName("help_button")
         self._handle_npe2_plugin(npe_version)
 
         if installed:
+            if is_conda_package(package_name):
+                self.source.setText('Conda')
             self.enabled_checkbox.show()
-            self.action_button.setText(trans._("uninstall"))
+            self.action_button.setText(trans._("Uninstall"))
```

Comment:
Would be nice to encapsulate all this toggling in a single function that takes as input a bool e.g. 'is_installed' and then toggles the values of the various widgets based on that bool e.g. using `setVisible(True/False)` rather than `show()/hide()`

Reply:
I tried to do this---put everything in one function, but the widget inside the collapsible widget was not playing well with setVisible.  I left it as is for now, because its working this way.